### ISSUE TITLE
Make completer menus reusable

### ISSUE THREAD

**Issue by Jaykul** [PowerShell/PSReadLine#1818]
Regarding the existing `MenuComplete` function (`Ctrl+Space`) and the new "F2" history menu, a couple of requests:

1. Can we expose these menus as _commands_ to make them usable by other modules as a way to prompt for input? It's always frustrating for PowerShell script authors that `$Host.UI.PromptForChoice` is so hard to use ...
2. Can we offer an command to turn **one** of these menus "on by default" to provide a slightly more intellisense-like feel to the prompt? Most people won't want that all the time, but it could be super useful on a temporary basis (particularly for learning new modules, since the `MenuComplete` completer shows the full syntax block for commands):
  ![image](https://user-images.githubusercontent.com/192942/93506803-c509a200-f8ea-11ea-8be0-4320d3e6a937.png)

**Comment by daxian-dbw** [comment]
For your comment (1), `MenuComplete` is part of PSReadLine, not the host. That means it won't be generally available to all applications that hosts PowerShell and thus won't be reliable for a module to use. Are you suggesting to move `MenuComplete` to PowerShell host?

**Comment by vexx32** [comment]
More sounds like the ask is to provide a PSReadLine cmdlet that allows users to invoke a MenuComplete (for F2) action for more general purposes, rather than just being a PSReadLine builtin functionality that is otherwise entirely inaccessible.

**Comment by daxian-dbw** [comment]
Right, but that depends on the PSReadLine module to be available, which may not be the case for a custom PowerShell host. So a module cannot use that to replace the `$Host.UI.PromptForChoice` call because the latter is part of the base type of the host, which is generally available.

**Comment by Jaykul** [comment]
> For your comment (1), `MenuComplete` is part of PSReadLine, not the host. That means it won't be generally available to all applications that hosts PowerShell and thus won't be reliable for a module to use. Are you suggesting to move `MenuComplete` to PowerShell host?

PowerShell scripts and modules can and do frequently take dependencies on (particular versions of) Powershell _modules_, but rarely if ever rely on specific versions of the PowerShell host.

I am suggesting that this should be a command available in the PSReadLine module which could be called by other modules or scripts which take a dependency on (a future version of) that module. Modules are versioned, and generally backward compatible  (and PSReadLine is, specifically, installable even down to Windows PowerShell), and are in my opinion _inherently_ the right shipping container for most functionality.

**Comment by vexx32** [comment]
Perhaps it'd be better if the UI functionality parts of PSRL is split out into a slightly more general-purpose module that can be used from PS and C# modules? That way, you could take a dependency on the UI portion without necessarily requiring PSRL itself to be loaded, and PSRL would just take that same dependency for the functionality it needs.

**Comment by Jaykul** [comment]
I don't think that's necessary, but I wouldn't argue against it. Remember though, that interactive "custom PowerShell hosts" that don't ship and **depend on** PSReadLine are _purely hypothetical_ in nature.

**Comment by totkeks** [comment]
Came here looking for a solution to have a not-so-ugly menu as provided by `PromptForChoice` or reimplementing the wheel.

Would be cool if this could be added as either a cmdlet provided by the PSReadline module or if possible, even to the PowerShell host.
